pagefaultgames/pokerogue
# [Bug] Fix ability Download switch in (1669)

## What are the changes?
1. Changes the Download ability.

## Why am I doing these changes?
1. Fixes: https://github.com/pagefaultgames/pokerogue/issues/1669

## What did change?
1. Check if the opponent has fainted yet before trying to check the stats for Download ability.
  -- This will no longer soft-lock if a Pokemon with Download switches in when all opponents have fainted.

## How to test the changes?
1. Create save state using overrides.ts:
```
export const STARTING_LEVEL_OVERRIDE: integer = 100;

export const MOVESET_OVERRIDE: Array<Moves> = [Moves.U_TURN];
```

2. Start the game with a Porygon as your second Pokemon (that will be switched into).
3. Use the move U-Turn with your level 100 Pokemon to faint the first wave opponent.
4. Switch into Porygon. The ability Download should not trigger and the game should continue.

## Checklist
